balloon-hash/src/balloon.rs

```diff
@@ -62,9 +69,16 @@ where
                 .map_with((params, secret), |(params, secret), (thread, memory)| {
                     hash_internal::<D>(pwd, salt, *secret, *params, memory, Some(thread))
                 })
-                .try_reduce(GenericArray::default, |a, b| {
-                    Ok(a.into_iter().zip(b).map(|(a, b)| a ^ b).collect())
-                })
+                .try_reduce(
+                    GenericArray::default,
+                    |mut a, #[cfg_attr(not(feature = "zeroize"), allow(unused_mut))] mut b| {
+                        a.iter_mut().zip(&b).for_each(|(a, b)| *a ^= b);
+                        #[cfg(feature = "zeroize")]
+                        b.zeroize();
+
+                        Ok(a)
```

Comment:
Here a better solution could look like this:
```rust
|mut a, b| {
    a.iter_mut().zip(&b).for_each(|(a, b)| *a ^= b);
    #[cfg(feature = "zeroize")]
    Zeroizing::new(b)
    Ok(a)
}
```

Reply:
:+1:
Is there something I should know about why this is better except skipping the whole `cfg` because of `mut`, or is that it?

Reply:
No, the `cfg_attr` simply looked a bit too noisy to me.

Reply:
Sadly `GenericArray` can't be wrapped into `Zeroizing` :(.